src/views/studio/MyVideosView/MyVideosView.tsx

```diff
@@ -0,0 +1,189 @@
+      sethasAnyVideos(true)
+    } else if (typeof totalCount === 'number' && hasAnyVideos === undefined) {
+      sethasAnyVideos(false)
+    }
+  }, [hasAnyVideos, totalCount])
+
+  // hook to tests draft should be deleted before final
+  useEffect(() => {
+    const createDrafts = async () => {
+      await removeAllDrafts()
+      for (let i = 0; i < 32; i++) {
+        await addDraft({
+          channelId: testChannelId,
+          title: i.toString(),
+          description: 'string',
+          isPublic: true,
+          hasMarketing: false,
+          isExplicit: false,
+        })
+      }
+    }
+    createDrafts()
+    return () => {
+      removeAllDrafts()
+    }
+  }, [])
+
+  const isDraftTab = currentTabName === 'Drafts'
+  const isLoading = loading || (videos?.length === 0 && (totalCount ?? 0) > 0)
+  const placeholderItems = Array.from({ length: isLoading ? videosPerPage : 0 }, () => ({
+    id: undefined,
+    progress: undefined,
+  }))
+  const videosWPlaceholders = [...(videos || []), ...placeholderItems]
+  const handleOnResizeGrid = (sizes: number[]) => setVideosPerRow(sizes.length)
+
+  const FetchMoreVideos = (page: number) =>
+    fetchMore({
+      variables: {
+        // substract 1 coz offset index starts at 0
+        offset: videosPerPage * (page - 1),
+        limit: videosPerPage,
+      },
+    })
+  // console.log({ videos, totalCount, hasAnyVideos, isLoading, isPublic_eq })
+  return (
+    <ViewContainer>
+      <StyledText variant="h2">My Videos</StyledText>
+
+      {hasAnyVideos === false ? (
+        <AddVideoView />
+      ) : (
+        <>
+          <TabsContainer>
+            <Tabs initialIndex={0} tabs={[...TABS]} onSelectTab={setCurrentTab} />
+          </TabsContainer>
+          {isDraftTab && (
+            // Should this really be dismissable?
+            <StyledDismissibleMessage
+              id="video-draft-saved-locally-warning"
+              title={'Video Drafts are saved locally'}
+              description={
+                'This mean you can only access one on the device you used to create it. Clearing your browser history will delete all your drafts.'
+              }
+            />
+          )}
+          <Grid maxColumns={null} onResize={handleOnResizeGrid}>
+            {!isDraftTab &&
+              videosWPlaceholders
+                // this makes for a smoother transition between pages
+                .slice(0, videosPerPage)
+                .map((video, idx) => (
+                  <VideoPreviewPublisher
+                    key={idx + '-' + currentTabName + '-' + currentPage}
+                    id={video.id}
+                    showChannel={false}
+                    isLoading={loading}
+                  />
+                ))}
+            {isDraftTab &&
+              drafts
+                // pagination slice
+                .slice(videosPerPage * (currentPage - 1), (currentPage - 1) * videosPerPage + videosPerPage)
```

Comment:
Maybe we should just keep the page 0-indexed and only add 1 to it when passing to the paginating component? Seems there's a lot of places where we subtract 1

Reply:
At the end we would have to do the same in the pagination component onChange / onMouseEnter i.e

```
          <PaginationContainer>
            <Pagination
              onChangePage={(page) => {
                setCurrentPage(page - 1)
                currentTabName !== 'Drafts' && fetchMoreVideos(page - 1)
              }}
              onMouseEnterPage={(page) => {
                currentTabName !== 'Drafts' && fetchMoreVideos(page - 1)
              }}
              page={currentPage + 1}
              itemsPerPage={videosPerPage}
              totalCount={isDraftTab ? drafts.length : totalCount}
            ></Pagination>
          </PaginationContainer>
```

Reply:
Hmm I think the `Pagination` component should actually be 0-indexed as well. I think we had discussion about this, right @drillprop ?

Reply:
yeah was my thought as well, even tho slightly confusing at first could be the best solution

Reply:
@eldiegod if I'm not wrong you can delete both `-1` from slice and everything should work the same. 
Tried by myself. Let me know if I'm missed something

Reply:
@eldiegod if I'm not wrong you can delete both -1 from slice and everything should work the same.
Tried by myself. Let me know if I'm missed something